pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -70,6 +71,21 @@ func (s *SecretGenerator) GenerateForDynakube(ctx context.Context, dk *dynakube.
 		if err != nil {
 			return errors.WithStack(err)
 		}
+
+		// TODO check if this can be refactored to be cleaner
+		// here we extract the token again from the 'endpoint.properties', which includes both the DT_METRICS_INGEST_URL and DT_METRICS_INGEST_API_TOKEN in a single string,
+		// while for the OTLP env vars we need the ingest token as a separate key within the secret.
+		// This way we do not have to look up the secret containing the token in the dynatrace namespace again (i.e. less kubernetes API calls),
+		// but this is also not ideal, as the extractApiToken relies on the endpoint.properties to be in a certain format.
+		// Ideally, we can refactor this to retrieve the endpoint and token at the beginning and then generate the content for the secrets,
+		// as they are required by the bootstrapper and the otlp env vars, respectively.
```

Comment:
you could use the "tokens" that is in the main reconciler https://github.com/Dynatrace/dynatrace-operator/blob/cee6877a2e4de9e12c8634ca9447fc7c9e6dd62d/pkg/controllers/dynakube/controller.go#L127-L128

This could be passed to the `bootstrapperconfig` as well, so it also doesn't query for them again. (separate improvment)